instagram/src/components/PostContainer/PostContainer.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import CommentSection from '../CommentSection/CommentSection';
+import styled from 'styled-components';
+import Username from '../../Styles/Username';
+
+// Styled Components
+const PostWrapper = styled.div`
+  max-width: 604px;
+  width: 100%;
+  margin: 0 auto 6rem;
+  border: 1px solid #dbdbdb;
+  border-radius: 5px;
+  background: #fff;
+`;
+
+const Header = styled.header`
+  display: flex;
+  align-items: center;
+  padding: 1.5rem;
+`;
+
+const Thumbmail = styled.img`
+  height: 30px;
+  width: 30px;
+  border-radius: 50%;
+  margin-right: 1rem;
+`;
+
+const ImageContainer = styled.div`
+  width: 100%;
+  margin-bottom: 1.5rem;
+`;
+
+const Instagram = styled.img`
+  width: 100%;
+  display: block;
+`;
+
+const IconContainer = styled.div`
+  padding: 0 1.5rem;
+  font-size: 2.5rem;
+  display: flex;
+
+  i {
+    margin-right: 1.5rem;
+    cursor: pointer;
+  }
+`;
+
+const Likes = styled.div`
+  padding: 0 1.5rem;
+  font-weight: bold;
+`;
+
+const PostContainer = ({ data, likePost, id, index }) => {
+  return (
+    <PostWrapper>
```

Comment:
Nice job with this! Personally I find this a big hard to read because I don't know what the semantic HTML is at a glance... but that might just be me. How do you like this approach?

Reply:
I like this approach so far but the css and html/jsx markup was already written so I felt really okay just refactoring them as styled components everywhere because I knew how the markup looked. So I'm not sure how I'll feel writing them from scratch... but I want to try.